native_client/deepspeech.h

```diff
@@ -1,137 +1,172 @@
+ * @param aLMPath The path to the language model binary file.
+ * @param aTriePath The path to the trie file build from the same vocabu-
+ *                  lary as the language model binary.
+ * @param aLMWeight The weight to give to language model results when sco-
+ *                  ring.
+ * @param aValidWordCountWeight The weight (bonus) to give to beams when
+ *                              adding a new valid word to the decoding.
+ */
+DEEPSPEECH_EXPORT
+void DS_EnableDecoderWithLM(ModelState* aCtx,
+                            char* aAlphabetConfigPath,
+                            char* aLMPath,
+                            char* aTriePath,
+                            float aLMWeight,
+                            float aValidWordCountWeight);
+
+/**
+ * @brief Use the DeepSpeech model to perform Speech-To-Text.
+ *
+ * @param aCtx The ModelState pointer for the model to use.
+ * @param aBuffer A 16-bit, mono raw audio signal at the appropriate
+ *                sample rate.
+ * @param aBufferSize The number of samples in the audio signal.
+ * @param aSampleRate The sample-rate of the audio signal.
+ *
+ * @return The STT result. The user is responsible for freeing the string.
+ */
+DEEPSPEECH_EXPORT
+char* DS_SpeechToText(ModelState* aCtx,
+                      short* aBuffer,
+                      unsigned int aBufferSize,
+                      int aSampleRate);
+
+/**
+ * @brief Setup a context used for performing streaming inference.
+ *        the context pointer returned by this function can then be passed
+ *        to {@link DS_FeedAudioContent()} and {@link DS_FinishStream()}.
+ *
+ * @param aPreAllocFrames Number of timestep frames to reserve. One timestep
+ *                        is equivalent to two window lengths (50ms), so
+ *                        by default we reserve enough frames for 3 seconds
+ *                        of audio.
+ * @param aSampleRate The sample-rate of the audio signal.
+ * @param[out] retval a context pointer that represents the streaming state. Can
+ *                    be null if an error occurs.
+ *
+ * @return Zero for success, non-zero on failure.
+ */
+DEEPSPEECH_EXPORT
+int DS_SetupStream(ModelState* aCtx,
+                   unsigned int aPreAllocFrames,
+                   unsigned int aSampleRate,
+                   StreamingState** retval);
+
+/**
+ * @brief Feed audio samples to an ongoing streaming inference.
+ *
+ * @param aCtx A streaming context pointer returned by {@link DS_SetupStream()}.
+ * @param aBuffer An array of 16-bit, mono raw audio samples at the
+ *                appropriate sample rate.
+ * @param aBufferSize The number of samples in @p aBuffer.
+ */
+DEEPSPEECH_EXPORT
+void DS_FeedAudioContent(StreamingState* aSctx,
+                         short* aBuffer,
+                         unsigned int aBufferSize);
+
+
+DEEPSPEECH_EXPORT
+char* DS_IntermediateDecode(StreamingState* aSctx);
```

Comment:
Do I have to free the returned pointer?

Reply:
Yes! Looks like I forgot to document this function, thanks for the heads up :)